lib_xunit_engine/src/main/x/xunit_engine/console/ConsoleExecutionListener.x

```diff
@@ -0,0 +1,82 @@
+import xunit_engine.Model;
+import xunit_engine.UniqueId;
+
+service ConsoleExecutionListener
+        implements ExecutionListener {
+
+    @Inject Console console;
+
+    private Map<UniqueId, Result> results = new HashMap();
+
+    Boolean success = True;
+
+    /**
+     * A singleton `Result` representing the initial result for a model.
+     */
+    static Result InitialResult = new Result(Successful, count=0);
+
+    @Override
+	void onStarted(Model model) {
+	    console.print($"Started:   {model.displayName}");
+	    results.put(model.uniqueId, InitialResult);
+	}
+
+    @Override
+	void onCompleted(Model model, Result result) {
+        UniqueId   id = model.uniqueId;
+        Exception? ex = result.exception;
+
+        this.success &= result.status.success;
+
+        if (model.isContainer) {
+            if (Result thisResult := results.get(id)) {
+                result = thisResult;
+            }
+        }
+        results.remove(id);
+
+        if (id.type == Iteration) {
+            assert id := id.parent();
+        }
+
+        if (UniqueId parentId := id.parent(), Result parentResult := results.get(parentId)) {
+            Result r = parentResult.merge(result);
+            results.put(parentId, r);
+        }
+
+	    if (result.status == Skipped) {
+	        if (ex.is(Exception)) {
+	            onSkipped(model, ex.message);
+	        } else {
+	            onSkipped(model, "");
+	        }
+	    } else {
+	        if (model.isContainer) {
+	            console.print($"Finished:  {model.displayName} {result.status} in {result.duration} (passed={result.succeeded} failed={result.failures} skipped={result.skipped} errors={result.errors})");
```

Comment:
There is no need to have such a long string:

	            console.print($|Finished: {model.displayName} {result.status} in {result.duration}\
	                           | (passed={result.succeeded} failed={result.failures}\ 
	                           | skipped={result.skipped} errors={result.errors})\
	                           );